test/finch/http2/pool_test.exs

```diff
@@ -92,7 +92,8 @@ defmodule Finch.HTTP2.PoolTest do
     :timer.sleep(10)
 
     # We can't send any more requests since the connection is closed for writing.
-    assert {:error, %{reason: :read_only}} = request(pool, req, [])
+    assert {:error, %Finch.Error{reason: :read_only}=error_read_only} = request(pool, req, [])
+    assert Exception.message(error_read_only)
```

Comment:
I'd keep it similar to other tests you added below. This means we'd no longer be testing our message/1 implementation but that's imho totally fine :)

```suggestion
    assert {:error, %Finch.Error{reason: :read_only}} = request(pool, req, [])
```